config/routes.rb

```diff
@@ -0,0 +1,108 @@
+Rails.application.routes.draw do
+
+  root 'users#index'
+
+  get  '/users/all_markets'  => 'users#all_markets', as: "all_markets"
```

Comment:
This view is really the markets index page. As such the route should probably be defined as:

``` get '/users/markets' => 'markets#index' ```

This would also require a change to your controller such that the market index method holds the logic that your "all_markets" method currently holds.